sklearn/preprocessing/_discretization.py

```diff
@@ -225,31 +237,35 @@ def fit(self, X, y=None):
                         n_samples, size=self.subsample, replace=False
                     )
                     X = _safe_indexing(X, subsample_idx)
-        elif self.strategy != "quantile" and isinstance(
-            self.subsample, numbers.Integral
-        ):
+
+        if self.strategy != "quantile" and isinstance(self.subsample, numbers.Integral):
```

Comment:
You should let this line how it was.

Reply:
I corrected this in https://github.com/scikit-learn/scikit-learn/pull/22048/commits/30edabcbd6efb87d38bfb47f004ca36a7cbea664 (l. 241)